src/ameba/rule/style/heredoc_escape.cr

```diff
@@ -0,0 +1,63 @@
+module Ameba::Rule::Style
+  # A rule that enforces using an escaped heredoc to escape interpolation or escape sequences instead of backslashes.
+  #
+  # For example, this is considered invalid:
+  #
+  # ```
+  # <<-DOC
+  #   This is an escaped \#{:interpolated} string \\n
+  #   DOC
+  # ```
+  #
+  # And should be written as:
+  #
+  # ```
+  # <<-'DOC'
+  #   This is an escaped #{:interpolated} string \n
+  #   DOC
+  # ```
+  #
+  # YAML configuration example:
+  #
+  # ```
+  # Style/HeredocEscape:
+  #   Enabled: true
+  # ```
+  class HeredocEscape < Base
+    include AST::Util
+
+    properties do
+      since_version "1.7.0"
+      description "Recommends using the escape heredoc start when escaping interpolation or other chars in a heredoc body"
+    end
+
+    MSG_ESCAPE_NEEDED     = "Use an escaped heredoc marker: `<<-'%s'`"
+    MSG_ESCAPE_NOT_NEEDED = "Unnecessary heredoc escape"
+
+    def test(source, node : Crystal::StringInterpolation)
+      return unless (code = node_source(node, source.lines)) && code.starts_with?("<<-")
+      # Heredocs without interpolation are always size 1
```

Comment:
This comment contradicts the final test case:

```cr
it "fails if an escaped heredoc doesn't contain interpolation"
```

Reply:
That testcase is checking if you have an escaped heredoc start that's unnecessary (by not having `#{}` or escape sequences)